bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableClusterUtilities.java

```diff
@@ -0,0 +1,222 @@
+   *
+   * @return the current state of the instance
+   */
+  public synchronized ListClustersResponse getClusters() {
+    logger.info("Reading clusters.");
+    String instanceNameStr = instanceName.getInstanceName();
+    ListClustersRequest listRequest =
+        ListClustersRequest.newBuilder().setParent(instanceNameStr).build();
+    return client.listCluster(listRequest);
+  }
+
+  /**
+   * Sets a cluster size to a specific size.
+   * @param clusterId
+   * @param zoneId
+   * @param newSize
+   * @throws InterruptedException
+   */
+  public synchronized void setClusterSize(String clusterId, String zoneId, int newSize)
+      throws InterruptedException {
+    Preconditions.checkArgument(newSize > 0, "Cluster size must be > 0");
+    Cluster cluster = getCluster(clusterId, zoneId);
+    int currentSize = cluster.getServeNodes();
+    if (currentSize == newSize) {
+      logger.info("Cluster %s already has %d nodes.", clusterId, newSize);
+    } else {
+      updateClusterSize(cluster.getName(), newSize);
+    }
+  }
+
+  /**
+   * @param clusterId
+   * @param zoneId
+   * @param incrementCount a positive or negative number to add to the current node count
+   * @return the new size of the cluster.
+   * @throws InterruptedException
+   */
+  public synchronized int incrementClusterSize(String clusterId, String zoneId, int incrementCount)
+      throws InterruptedException {
+    Preconditions.checkArgument(incrementCount != 0,
+      "Cluster size cannot be incremented by 0 nodes. incrementCount has to be either positive or negative");
+    Cluster cluster = getCluster(clusterId, zoneId);
+    if (incrementCount > 0) {
+      logger.info("Adding %d nodes to cluster %s", incrementCount, clusterId);
+    } else {
+      logger.info("Removing %d nodes from cluster %s", -incrementCount, clusterId);
+    }
+    int newSize = incrementCount + cluster.getServeNodes();
+    updateClusterSize(cluster.getName(), newSize);
+    return newSize;
+  }
+
+  /**
+   * Update a cluster to have a specific size
+   *
+   * @param clusterName The fully qualified clusterName
+   * @param newSize The size to update the cluster to.
+   * @throws InterruptedException
+   */
+  private void updateClusterSize(String clusterName, int newSize) throws InterruptedException {
+    logger.info("Updating cluster %s to size %d", clusterName, newSize);
+    Operation operation = client
+        .updateCluster(Cluster.newBuilder().setName(clusterName).setServeNodes(newSize).build());
+    waitForOperation(operation.getName(), 30);
+    logger.info("Done updating cluster %s.", clusterName);
+  }
+
+  /**
+   * Waits for an operation like cluster resizing to complete.
+   * @param operationName The fully qualified name of the operation
+   * @param maxSeconds The maximum amount of seconds to wait for the operation to complete.
+   * @throws InterruptedException if a user interrupts the process, usually with a ^C.
```

Comment:
If you start a Dataflow job, and the resizing somehow stalls, a user would send an interrupt.